File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/dashboards/widgets/ValidationStabilityChartWidget.kt
package net.nemerosa.ontrack.model.dashboards.widgets

import org.springframework.stereotype.Component

@Component
class ValidationStabilityChartWidget : AbstractWidget<ValidationChartWidgetConfig>(
    key = "home/ValidationStabilityChart",
    name = "Validation stability",
    description = "Chart displaying the percentage of builds being validated.",
    defaultConfig = ValidationChartWidgetConfig(project = null, branch = null, validationStamp = null),
    preferredHeight = 30,
)

File: ontrack-web-core/components/widgets/home/ValidationStabilityChartWidget.js
import ValidationStampStabilityChart from "@components/validationStamps/ValidationStampStabilityChart";
import ValidationChartWidget from "@components/widgets/home/ValidationChartWidget";

export default function ValidationStabilityChartWidget(props) {
    return (
        <ValidationChartWidget
            prefix="Stability of"
            chart={ValidationStampStabilityChart}
            {...props}
        />
    )
}
